PMSCore/Permission.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PMSCore
+{
+
+    /// <summary>
+    /// Represents a user's permission with a level and description.
+    /// </summary>
+    public class Permission
+    {
+        /// <summary>
+        /// Gets the level of the permission.
+        /// A higher value indicates more access or privileges.
+        /// </summary>
+        public int PermissionLevel { get; private set; }
+
+        /// <summary>
+        /// Gets the description of the permission.
+        /// Provides additional context about what the permission level entails.
+        /// </summary>
+        public string PermissionDescription { get; private set; }
```

Comment:
private set?

Reply:
Changed implementation in newest update. I might be misunderstanding, but if we're using the permission as a DI shouldn't we have it as immutable outside of the constructor? Sorry if I did misunderstand. Lmk what you think when you see the updated implementation. :)